feat(issue-details): Refactor header to grid

leeandher opened the pull request:
- Reduces foreheader (hehe)
- Adds tooltip for longer values
- New styling for big counts
- Adds simple css transition for resolve/archive
- Remove redundant styles and elements


https://github.com/user-attachments/assets/dc1b67a5-b785-4ddd-a7b7-56b368680a65



Before:

![image](https://github.com/user-attachments/assets/23a50604-d713-479e-94fc-87ad929a4272)

After:

![image](https://github.com/user-attachments/assets/2f5a7561-87b7-47a0-b069-b251424198b9)

roggenkemper commented on static/app/views/issueDetails/streamline/header.tsx:
```diff
@@ -76,245 +77,191 @@ export default function StreamlinedGroupHeader({
+    background: linear-gradient(
+      to right,
+      ${p => p.theme.background},
+      ${p => Color(p.theme.success).lighten(0.5).alpha(0.15).string()}
```
someone suggested adding a different color for archived (blue was the suggestion), this might be a good time to try that out?

leeandher replied:
good point, lemme try that out

roggenkemper commented on static/app/views/issueDetails/streamline/header.tsx:
```diff
@@ -76,245 +77,191 @@ export default function StreamlinedGroupHeader({
+          ]}
+        />
+        <HeaderGrid>
+          <Flex gap={space(0.75)} align="flex-end">
```
the secondary title doesn't look aligned with the primary title (too low)

leeandher replied:
That's intentional from the designs: 
![image](https://github.com/user-attachments/assets/17fb675c-e22e-4eec-80fc-0d59e52631ba)

How it looks is another thing though 🤷 I don't mind it, since the primary title font is larger

roggenkemper replied:
hmm , in the designs it seems like the bottom of each letter is aligned? but in the preview the secondary title looks like it's below

leeandher commented:
> Looks like some primary titles are getting cut off <img alt="Screenshot 2024-10-21 at 1 48 14 PM" width="621" src="https://private-user-images.githubusercontent.com/46740234/378593742-1ebb89cb-00de-4a89-9f19-6a447680823e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NDQxMzksIm5iZiI6MTcyOTU0MzgzOSwicGF0aCI6Ii80Njc0MDIzNC8zNzg1OTM3NDItMWViYjg5Y2ItMDBkZS00YTg5LTlmMTktNmE0NDc2ODA4MjNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDIwNTAzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhN2VjNDM5OTNiYzAwZmM2Mjc1NmQ1MWU1ZDU0OGJlNDk4NTBmZDFiZjMyNDEwMTJlNTlmYzRlYmQ3YjIzZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1WO8yvCGAjJayGpxvLffduYffyjJ1P5Wk7Rncezl8aE">

oh no 🫨 i'll take a look

scttcper commented:
i think the title looks better on small sizes with grid, but we're getting some overflow here on assignee etc

before
![image](https://github.com/user-attachments/assets/7a6e090c-5282-4d83-99c9-3782db780b73)

after
![image](https://github.com/user-attachments/assets/d2e035b1-db71-4957-b108-e5af513fc82e)

leeandher commented:
> i think the title looks better on small sizes with grid, but we're getting some overflow here on assignee etc

Addressed for smaller screen sizes. Also changed the actions flickering in and out at different screen sizes.

malwilley commented:
@leeandher for the issue stream layout changes, we switched the position of the message and the subtitle. Is it possible for us to match the new stream layout for issue details? Otherwise it might be jarring if you click from the issue stream and see the title order change.

Instead of "Error   /checkout" we now show "Error 500 - Internal Server Error"

leeandher commented:
> @leeandher for the issue stream layout changes, we switched the position of the message and the subtitle. Is it possible for us to match the new stream layout for issue details? Otherwise it might be jarring if you click from the issue stream and see the title order change.
> 
> Instead of "Error /checkout" we now show "Error 500 - Internal Server Error"

@malwilley Switched it!
